msgio.go

```diff
@@ -200,7 +204,10 @@ func (sem *semaphore) enable() {
 }
 
 func (sem *semaphore) lock() {
```

Comment:
couldn't the same functionality be done by requiring `lock` to take an additional parameter `ctx context.Context`?

after a brief scan of the source code, it seems to me all locations where `semaphore.lock` is called also take a `context.Context` as argument.

Reply:
I agree that's a better approach, will update to use context passed in from argument instead.